### PR TITLE
Update git guidelines and branch management.

### PR DESCRIPTION
**Pull request by mvandeberg**

### CODE REVIEW

File: doc/git-guidelines.md [hunk 1]
```diff
@@ -1,105 +1,134 @@
...
-  from `master` has risks.  We recommend that any block producing node
-  build from `stable`. If you want to test new features, develop is the
-  correct branch.
+- `stable`: Points to the latest version of code that is production-ready, and has been tested in production. This is the branch we recommend for exchanges.
+- `0.21.x`: This is the current release branch and will always point to the most recent release. Witnesses and seed node operators may want to run this branch. `stable` and `0.21.x` may diverge if there are minor bug fixes or witness only security fixes that does not need exchanges to update for. 
+- `master`: The active development branch. We will strive to keep `develop` in a working state. All PRs must pass automated tests before being merged. While this is not a guarantee that `master` is bug-free, it will guarantee that the branch is buildable in our standard build configuration and passes the current suite of tests. That being said, running a node from `master` has risks.  We recommend that any block producing node build from `stable`. If you want to test new features, develop is the correct branch.
```
Comment (sgerbino) [inline doc/git-guidelines.md]: Should be, ... "We will strive to keep `master` in a working state."

Comment (sgerbino) [reply doc/git-guidelines.md]: > We recommend that any block producing node build from `stable`. If you want to test new features, develop is the correct branch.

Probably should be:

We recommend that any block producing node build from the latest tag. If you want to test new features, `master` is the correct branch.

File: doc/git-guidelines.md [hunk 2]
```diff
@@ -1,105 +1,134 @@
...
-All pull requests should reference the issue(s) they relate to in order to
-create a chain of documentation.
-
-If your pull request has merge conflicts, rebase against `origin/develop`,
-resolve the merge conflicts, force push to your branch, and resubmit the
-pull request.
+All changes to `0.21.x` and `master` are done through GitHub Pull Requests (PRs). This is done for several reasons:
+
+- It enforces two factor authentication. GitHub will only allow merging of a pull request through their interface, which requires the dev to be logged in.
+- If enforces testing. All pull requests undergo automated testing before they are allowed to be merged.
```
Comment (sgerbino) [inline doc/git-guidelines.md]: > If enforces testing.

It enforces testing.

File: doc/git-guidelines.md [hunk 3]
```diff
@@ -1,105 +1,134 @@
...
+- It enforces best practices. Because of the cost of a pull request, developers are encouraged to do more testing themselves and be certain of the correctness of their solutions.
+- If enforces code review. All pull requests must be reviewed by a developer other than the creator of the request. Pull requests made by external developers should be reviewed by two internal developers. When a developer reviews and approves a pull request they should +1 the request or leave a comment mentioning their approval of the request. Otherwise they should describe what the problem is with the request so the developer can make changes and resubmit the request.
```
Comment (sgerbino) [inline doc/git-guidelines.md]: > If enforces code review.

It enforces code review.

**Comment by jredbeard** [comment]
I agree with these changes to git guidelines and branch management 👍

**Pull request merged**